src/txn/txn.c

```diff
@@ -1985,6 +1991,7 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     return (0);
 
 err:
+    WT_ASSERT_ALWAYS(session, !logged, "logged the transaction but commit failed.");
```

Comment:
Instead of this assert always, how about adding an if check similar to `cannot_fail` or integrating it with the same check with additional information if it is logged to keep the failure point at one location?